vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -593,6 +610,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	case types.VirtualMachinePowerStateSuspended:
 		d.Set("power_state", "suspended")
 	}
+	
+	// Check for the presence of a virtual Trusted Platform Module (TPM) device.
+	var vtpmPresent bool
+	for _, dev := range vprops.Config.Hardware.Device {
+		if _, ok := dev.(*types.VirtualTPM); ok {
+			vtpmPresent = true
+			break
+		}
+	}
+	_ = d.Set("vtpm_present", vtpmPresent)
```

Comment:
```suggestion
	var isVTPMPresent bool
	for _, dev := range vprops.Config.Hardware.Device {
		if _, ok := dev.(*types.VirtualTPM); ok {
			isVTPMPresent = true
			break
		}
	}
	_ = d.Set("vtpm_present", isVTPMPresent)
```

A bit more idiomatic as it clearly indicates that it is a boolean variable.

Another option, is `isVTPMFound`.